lib/analyzer/image-inspector.ts

```diff
@@ -71,21 +73,37 @@ async function getArchiveLocation(
   ) {
     hostname = "registry-1.docker.io";
```

Comment:
Wondering if there's some way (or value in doing it?) to have a function capture all of this for us, so we can simply do:
```js
const { registryHostname, remainder /* not sure what this is a remainder of? */, imageName, imageTag } = extractDetailsFromTargetImage(targetImage);
```